tests/test_page.py

```diff
@@ -10,23 +10,48 @@
 
 
 def test_base_url(base_url, page):
-    assert base_url == page.url
+    assert base_url == page.canonical_url
 
 
-def test_url_keywords(base_url, selenium):
+def test_canonical_url_absolute(base_url, selenium):
+    url_template = 'https://www.test.com/'
+
+    class MyPage(Page):
+        URL_TEMPLATE = url_template
+    page = MyPage(selenium, base_url)
+    assert url_template == page.canonical_url
+
+
+def test_canonical_url_empty(selenium):
+    page = Page(selenium)
+    assert page.canonical_url == ''
+
+
+def test_canonical_url_keywords(base_url, selenium):
     value = str(random.random())
 
     class MyPage(Page):
-        _url = '{key}'
-    page = MyPage('foo', selenium, key=value)
-    assert value == page.url
+        URL_TEMPLATE = '{key}'
+    page = MyPage(selenium, base_url, key=value)
+    assert base_url + value == page.canonical_url
+
+
+def test_canonical_url_prepend(base_url, page, selenium):
+    assert base_url == page.canonical_url
```

Comment:
Well, it's appending the base URL to '' isn't it? Although I agree this is tested in `test_canonical_url_keywords` I think it's worth having a test with a base URL and a simple path.